packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -2285,10 +2293,47 @@ function removeEventListenerFromChild(
   return false;
 }
 // $FlowFixMe[prop-missing]
-FragmentInstance.prototype.focus = function (this: FragmentInstanceType) {
-  traverseFragmentInstance(this._fragmentFiber, setFocusIfFocusable);
+FragmentInstance.prototype.focus = function (
+  this: FragmentInstanceType,
+  focusOptions?: FocusOptions,
+): void {
+  traverseFragmentInstance(
+    this._fragmentFiber,
+    setFocusIfFocusable,
+    focusOptions,
+  );
+};
+// $FlowFixMe[prop-missing]
+FragmentInstance.prototype.focusLast = function (
+  this: FragmentInstanceType,
+  focusOptions?: FocusOptions,
+) {
+  traverseFragmentInstanceReverse(
```

Comment:
Alternatively this could be a bit simpler to reuse the `traverseFragmentInstance` to collect the child nodes and then loop through them applying focus. The tradeoff would be an array allocation and an additional pass through the list, though I don't expect this to get hit too hard and we'd likely break early on the second loop.

Reply:
Mainly for code size I think it's probably worth the array allocation in this case since it's a very rare case that doesn't pay for itself.

The other reason is that if this is a very long list of items, then you might hit stack overflow since you need a recursive function for each sibling to allocate the temporary memory anyway. If we had a reverse pointers it would be another story but since we don't it could be probably.